src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
```

Comment:
accoding to changes we did on level higher we need to update Props type here
```suggestion
type Props = {
  userSelectedId: string;
  selectPost: (postId?: number) => void;
  selectedPostId?: number;
};
```

Reply:
also on level higher `userSelectedId` is called `currentUser`
I would go with `selectedUserId` to stay persistent in both components and also comparing to `selectedPostId` naming